Dev

iatenine opened the pull request:
Look at these 193 commits I've made!

redn6gx committed:
add ShopProduct class to models

redn6gx committed:
add Product class to models

redn6gx committed:
add properties to Product model

redn6gx committed:
Add properties to ShopProduct model

redn6gx committed:
Add GeneratedValue annoation to Produc pk

redn6gx committed:
Add entity annotation and names tables

redn6gx committed:
Remove ShopProduct fk from Product model

redn6gx committed:
Add Product fk to ShopProduct model

David427 committed:
Created User/Customer/Seller models and repos

redn6gx committed:
Add Category model

redn6gx committed:
Add GeneratedValue to Category pk

John-Salguero committed:
Wrote tests for JWT validation

David427 committed:
Started to create H2 database

ntluandt95 committed:
add test case for login endpoit

John-Salguero committed:
Finished up writing tests for validation of JWT Token

* Created and tested method validate for jwtTokenUtil Class
* The class and method still need to be implemented

David427 committed:
Updated model classes and H2 configuration

dscanga33 committed:
Added a create user function in controller and service layers, as well as tests for each. AppConfig added to implement password hashing

John-Salguero committed:
The Tests are not working because JUnit cannot find the
test classes in the classpath. I don't know what it
wrong with the class path.

benkarlsberg committed:
adjust column fields and add h2 application.properties and test placeholders

benkarlsberg committed:
add @column fields to models

John-Salguero committed:
Added application.properties, and now everything builds and is found

redn6gx committed:
Create feature-product-display-bobby branch

benkarlsberg committed:
add ProductServicesTests skeleton

benkarlsberg committed:
fixed categoryList name

benkarlsberg committed:
add tests for product service layer

David427 committed:
Completed UserRepoTests

David427 committed:
Merge remote-tracking branch 'origin/register-scanga' into register-david

# Conflicts:
#	src/main/java/com/revature/models/User.java

redn6gx committed:
Add getProductById test to ProductControllersTests

redn6gx committed:
Add not found check to testGetProductById

redn6gx committed:
Add ProductIntegrationTest class

redn6gx committed:
Add findProductById test

redn6gx committed:
Add ProductServiceIntegrationTest

redn6gx committed:
Add testGetProductById to ProductServiceIntegrationTest

redn6gx committed:
Add ProductController

redn6gx committed:
Add GET request method getProductById

Anthony-Davis committed:
testing for login jwtFilter

benkarlsberg committed:
add category service test getCategoryById()

benkarlsberg committed:
fixed variable naming

Milanoboy20 committed:
feat: added ShopProductIntegrationTest(all passed)

Milanoboy20 committed:
feat: added Shop Product Services

benkarlsberg committed:
commented out CategoryControllersTests and implement Product Service

benkarlsberg committed:
implement ProductRepo

benkarlsberg committed:
refactor ProductRepo to ShopProductRepo

John-Salguero committed:
Fixed malformed application.properties file

John-Salguero committed:
Merge remote-tracking branch 'origin/login-thanh' into feature-login

ramirez45 committed:
"test for loadUserByUsername"

Cacealeos committed:
ShopProduct Test and Shop Product Controller  test

David427 committed:
Updated models, implemented Service/Controller

ntluandt95 committed:
commit before merge to david branch to get user model

ntluandt95 committed:
Merge branch 'register-david' into login-thanh

John-Salguero committed:
Adding Implementation of JwtTokenUtility

* Using environemnet variables defined in application.properties to
determing secret and issuer
* The happy test fails needs to be debugged

John-Salguero committed:
Merge remote-tracking branch 'origin/login-Candido' into feature-login

John-Salguero committed:
Tests now work, JwtTokenUtility is now good to go

* Fixed the hard coded token testing against expiration

John-Salguero committed:
Merge branch 'login-john' into feature-login

David427 committed:
Created Customer/Seller Services and Controllers

Anthony-Davis committed:
testing JwtFilter

* started mocking dependencies for filter
* testing invalid token for filter

John-Salguero committed:
Merge branch 'login-john' into login-anthony

ntluandt95 committed:
before merge with john

redn6gx committed:
Remove unecessary comments

ntluandt95 committed:
merge conflict resolved

ntluandt95 committed:
merge conflict resolved

David427 committed:
Updated controllers

Anthony-Davis committed:
testing JwtFilter

* commit changes to merge with registration to obtain userRepo interface

benkarlsberg committed:
fixed tests and checked for passing

Milanoboy20 committed:
fix: integrated model classes with all tests passed.

John-Salguero committed:
Merge remote-tracking branch 'origin/register-david' into login-anthony

John-Salguero committed:
Merge remote-tracking branch 'origin/login-thanh' into feature-login

redn6gx committed:
Fix constructor for testProduct

redn6gx committed:
Add EntityScan and EnableJpaRepositories annotations to DartCartApplication

redn6gx committed:
Add missing ProductService import

redn6gx committed:
Add missing ProductRepo import

redn6gx committed:
Add missing ProductService import

redn6gx committed:
Add missing application context to DartCartApplicationTests

redn6gx committed:
Hard code primary keys and change double quotes to single quotes

redn6gx committed:
Hard code primary key for category

redn6gx committed:
Set web environment in SpringBoot Test Annotation

John-Salguero committed:
Refactored all the merge conflicts for login-feature

* Fixed Typos in the packages
* Refactored it to reflect the changes in the register feature

John-Salguero committed:
Merge branch 'login-anthony' into login-john

* Made sure the merge with register class happens effortlessly

John-Salguero committed:
Implemented the Generate Token and Generate Token Tests

* Added tests for the JwtTokenUtil generate token
* Fully implemented the gerate token method

Milanoboy20 committed:
fix: removed discount tests from integration test

Milanoboy20 committed:
fix: findShopProductById method returns optional

Commit:
tommy's changes

ramirez45 committed:
"test for loadUserByUsername"

Commit:
Merge branch 'feature-product-display-tommy' of https://github.com/Revature-DartCart/DartCart-Server into feature-product-display-tommy

Commit:
added product repo and shopproduct controller to main

Commit:
recent changes

redn6gx committed:
Reimplement method with Optionals

redn6gx committed:
Reimplement testGetProductById with Optionals

redn6gx committed:
Rename variables to match formatting of other tests

redn6gx committed:
Reimplement testGetProductById to use optionals

redn6gx committed:
Make mock Product static

redn6gx committed:
Separate testGetProductById into two different tests for success and fail cases

redn6gx committed:
Refactor ProductService to ProductServiceImpl

redn6gx committed:
Fix the error in testGetProductById where a variable can be null

iatenine committed:
Standardized and Consolidated Models

Created a standardized models package, consolidated the Customer and User tables together, applied naming conventions and
fixed an inaccurate ID from the old methods thanks to contributions by Robert Davis. Branch is based off of dev as to ensure
minimal conflicts and package structure was designed to reduce merge conflicts. Most developers should be able to merge this
branch into theirs with no conflicts so long as they first delete any models package they already have

iatenine committed:
Hotfix: Get drivers outside of the test folder

ntluandt95 committed:
fix test cases

David427 committed:
Incorporated updated models

benkarlsberg committed:
fixed Service methods and associated tests and removed commented code

ramirez45 committed:
"loadUserByUsername method implementation"

Commit:
fixed repo error

Commit:
fixed errors

Commit:
remove models pkg

Commit:
dartcartapp gone

ntluandt95 committed:
using authentication manager

ramirez45 committed:
Merge remote-tracking branch 'origin/feature-login' into login-Candido

ramirez45 committed:
"loadUserByUsername method implementation"

Anthony-Davis committed:
testJwtFilter changes

added sad and happy test cases

Anthony-Davis committed:
tests for JwtFilter

written happy and sad cases for jwtFilter

ryhaque committed:
getUserNameTests

John-Salguero committed:
Merge remote-tracking branch 'origin/login-thanh' into feature-login

John-Salguero committed:
Refactored Configuration Files

Anthony-Davis committed:
JwtTokenFilter

jwtFilter tests updated.
implementation added in utilities/JwtTokenFilter

David427 committed:
Updated models to comply with new structure

John-Salguero committed:
Merge remote-tracking branch 'origin/login-leland_ryan' into feature-login

Anthony-Davis committed:
Merge remote-tracking branch 'origin/feature-login' into login-anthony

ryhaque committed:
check user name updated

Milanoboy20 committed:
fix: using base models now

ryhaque committed:
Merge pull request #12 from Revature-DartCart/login-leland_ryan

check user name updated

John-Salguero committed:
Merge remote-tracking branch 'origin/login-thanh' into feature-login

* Thanh's Tests still are failing

John-Salguero committed:
Merge remote-tracking branch 'origin/login-anthony' into feature-login

John-Salguero committed:
Fixed the Tests for JwtFilter

John-Salguero committed:
Merge remote-tracking branch 'origin/feature-register' into feature-login

* Merged Register feature branch into the login feature branch
* only 2 tests remain to implement

John-Salguero committed:
Got the Login Tests working

John-Salguero committed:
Login Functionality now communication with Front-End

* Added a CORS Filter Configuration
* Changed the default port of the server

David427 committed:
Updated models to comply with new structure

redn6gx committed:
Change product from type int to type Product

redn6gx committed:
Split pass and fail cases into two different tests

redn6gx committed:
Change casing of Shop_product to shop_product

redn6gx committed:
Change ShopProductControllerTests to handle optionals

redn6gx committed:
Add mock ShopProduct

redn6gx committed:
Add important annotations to ShopProductControllersTests

redn6gx committed:
Remove unecessary comment

redn6gx committed:
Add SpringBootTest annotation

redn6gx committed:
Rename ShopProductRepository to ShopProductRepo

redn6gx committed:
Add ShopProduct mock object

redn6gx committed:
Change getShop_ProductById to handle optionals

redn6gx committed:
Rename getShop_ProductById to getShopProductById

redn6gx committed:
Rename product to ShopProduct

redn6gx committed:
Add assertion statements

redn6gx committed:
Split pass and fail cases in getShopProductById into two tests

redn6gx committed:
Add EntityScan and EnableJpaRepositories annotations to DartCartApplication

redn6gx committed:
Remove parameters from test methods

redn6gx committed:
Remove category test files

Commit:
added the Optionals

David427 committed:
Need to implement JWT functionality on UserController

dscanga33 committed:
Merge remote-tracking branch 'origin/feature-login' into register-scanga

# Conflicts:
#	pom.xml

dscanga33 committed:
/Register returns a token

Milanoboy20 committed:
feat: added data.sql

redn6gx committed:
Add missing tests for getAllShopProducts. All tests pass.

ramirez45 committed:
"modification to AuthServiceTest"

redn6gx committed:
Merge pull request #9 from Revature-DartCart/feature-product-display-bobby

Completed ProductController, ProductControllerTests, ProductIntegrationTests and ProductServiceIntegrationTests

redn6gx committed:
Merge pull request #5 from Revature-DartCart/feature-product-display-JHerr

ShopProduct Test and Shop Product Controller  test

redn6gx committed:
Merge branch 'feature-product-display' into product-display-abdul

redn6gx committed:
Merge pull request #1 from Revature-DartCart/product-display-abdul

Completed Tasks

John-Salguero committed:
Login Feature Fully complete on Back-End

* Renamed the test Classes for Uniformity
* Merged with Candido's finished branch
* Double Checked all Test Cases
* Changed the Default Port to 9005

redn6gx committed:
fix merging conflicts

redn6gx committed:
Merge pull request #4 from Revature-DartCart/feature-product-display-tommy

Feature product display tommy

redn6gx committed:
Fix merging conflicts

redn6gx committed:
Merge pull request #10 from Revature-DartCart/feature-product-display-ben

Feature product display ben

iatenine committed:
Modified gitignore to account for DS_Store files

dscanga33 committed:
Register returns a user

iatenine committed:
Removed unnecessary comments and printlns

iatenine committed:
Converted to Optionals and stuff

iatenine committed:
Update Tests

iatenine committed:
Merge pull request #13 from Revature-DartCart/feature-login

Feature login

iatenine committed:
Fixed tests to meet SQL data constraint rules

iatenine committed:
Merge branch 'feature-register' of github.com:Revature-DartCart/DartCart-Server into feature-register

iatenine committed:
Merge conflicts + remove DS_Store files

iatenine committed:
Merge branch 'dev' into feature-register

iatenine committed:
Merge pull request #11 from Revature-DartCart/feature-register

Feature register

redn6gx committed:
Change project structure to meet new conventions

redn6gx committed:
Merge pull request #14 from Revature-DartCart/feature-product-display-conventions

Change project structure to meet new conventions

John-Salguero committed:
minor changes for frontend

* Changed the AuthController return type to return full user
* Added all h2 endpoints to be permited without authentication

John-Salguero committed:
Merge branch 'feature-login' of workgithub:Revature-DartCart/DartCart-Server into feature-login

RichardSimpson235 committed:
Fix bug with data.sql trying to insert into ShopProduct table

When trying to perform INSERT INTO ShopProducts with @Table(name = "ShopProducts") on the class, Spring Boot would be unable to load the application context, saying that it could not find the ShopProducts table. Turns out this happens because Hibernate's default NamingStrategy automatically converts camel case and pascal case into snake case. Changing the sql to Shop_Products fixed the issue.

iatenine committed:
Merge dev

iatenine committed:
Merged more stuff.... Richard...

RichardSimpson235 committed:
Merge pull request #15 from Revature-DartCart/feature-product-display

Feature product display

Milanoboy20 committed:
feat: feature-search branch created

RichardSimpson235 committed:
Merge pull request #16 from Revature-DartCart/feature-login

Feature login

Milanoboy20 committed:
feat: add search function to ShopProductServiceImpl, all tests passed

RichardSimpson235 committed:
Fix minor issues with configuration and add quantity field ot OrderDetails

iatenine committed:
Merge pull request #17 from Revature-DartCart/minor-fix-config-props

Fix minor issues with configuration and add quantity field ot OrderDe…

Milanoboy20 committed:
feat: add search controller function, all tests pass

David427 committed:
Updated from dev

David427 committed:
Implemented Seller registration and Shop creation

Milanoboy20 committed:
fix: add documentation for searchShopProducts function

iatenine committed:
Merge pull request #20 from Revature-DartCart/feature-register

Feature register

iatenine committed:
Merge pull request #21 from Revature-DartCart/feature-search

SearchProducts Function and tests(all passed)

iatenine committed:
Fix: Ant Matchers

Add Ant Matcher for /shop_products to fix /display route (needs to have route corrected on client side)

iatenine committed:
Removed data.sql

RichardSimpson235 committed:
Merge pull request #22 from Revature-DartCart/fix/ant-configs-for-display-products

Fix/ant configs for display products

RichardSimpson235 commented on src/main/java/com/revature/models/OrderDetail.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Column(length = 1000, updatable = false)
+    private String description;
+
```
Need to move quantity here

RichardSimpson235 commented on src/main/java/com/revature/models/OrderDetails.java:
```diff
@@ -0,0 +1,32 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+public class OrderDetails {
```
Need to delete this to use other class

RichardSimpson235 commented on src/main/resources/application.properties:
```diff
@@ -1 +1,13 @@
+spring.h2.console.path=/h2
+spring.jpa.defer-datasource-initialization=true
 
+
```
Think we need to delete this since the main branch uses a different configuration

RichardSimpson235 reviewed (changes requested):
Just a couple of things that need fixing

iatenine committed:
Fix: Removed redundancies

Removed old OrderDetails model in favor of OrderDetail
Removed SellerController with no routes contained

RichardSimpson235 committed:
Merge pull request #26 from Revature-DartCart/fix/remove-redundancies

Fix/remove redundancies
